Schema replication example

DanielePalaia opened the pull request:
This closes #
Added an example to use the schema replication plugin using the topology operator part of:
https://github.com/rabbitmq/service-operator-experience/issues/46

**Note to reviewers:** remember to look at the commits in this PR and consider if they can be squashed
**Note to contributors:** remember to re-generate client set if there are any API changes

## Summary Of Changes

## Additional Context

Zerpet reviewed (commented):
It looks well overall. Here are a few points that we may revisit to improve the experience of using this example:

- Using a load balancer service type is not always possible e.g. kind
- This example requires manual editing the `schema-replication-*` files with a real IP
- Namespaces `upstream` and `downstream` must already exist

If we deploy both `RabbitmqCluster` in the same cluster, we can leverage Kubernetes internal DNS. Then we can declare the endpoints as `<instance-name>.<namespace>.svc` i.e. `upstream.upstream.svc:5672`. This addresses the first two pain points.

Regarding the namespaces, we could include a `Namespace` object in the YAML manifests, or make a comment in the readme.

For the purpose of demonstrating the capabilities of the plugin and the Topology Operator, deploying in the same Kubernetes cluster and different namespaces is acceptable.

DanielePalaia commented:
@Zerpet this is fine. I modified the endpoints of the SchemaReplication and I also renamed upstream and downstream clusters in upstream-cluster and downstream-cluster in order to differentiate the names from the namespaces.
For the first point, as this should run on a commercial versions the users needs also to setup secrets for the image registry after creating the namespaces, so I documented it in the README. What do you think?

DanielePalaia commented:
Closed as it was discussed to just document it in the commercial version.
